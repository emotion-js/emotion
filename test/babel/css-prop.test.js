import * as babel from 'babel-core'
import plugin from '../../src/babel'
import * as fs from 'fs'

jest.mock('fs')

fs.existsSync.mockReturnValue(true)

describe('babel css prop', () => {
  test('basic with extractStatic', () => {
    const basic = '(<div className="a" css={`color: brown;`}></div>)'
    const { code } = babel.transform(basic, {
      plugins: [[plugin, { extractStatic: true }]],
      filename: __filename,
      babelrc: false
    })
    expect(code).toMatchSnapshot()
    expect(fs.writeFileSync).toHaveBeenCalledTimes(1)
    expect(fs.writeFileSync.mock.calls[0][1]).toMatchSnapshot()
  })

  test('basic inline', () => {
    const basic = '(<div className="a" css={`color: brown;`}></div>)'
    const { code } = babel.transform(basic, {
      plugins: [plugin]
    })
    expect(code).toMatchSnapshot()
  })

  test('basic object', () => {
    const basic = '(<div className="a" css={{ color: \'brown\' }}></div>)'
    const { code } = babel.transform(basic, {
      plugins: [plugin]
    })
    expect(code).toMatchSnapshot()
  })

  test('dynamic inline', () => {
    const basic = `(<div className="a" css={\`color: $\{color};\`}></div>)`
    const { code } = babel.transform(basic, {
      plugins: [plugin]
    })
    expect(code).toMatchSnapshot()
  })

  test('no css attr', () => {
    const basic = '(<div></div>)'
    const { code } = babel.transform(basic, {
      plugins: [plugin]
    })
    expect(code).toMatchSnapshot()
  })

  test('with spread arg in jsx opening tag', () => {
    const basic = '(<div className="a" css={`color: brown;`} {...rest}></div>)'
    const { code } = babel.transform(basic, {
      plugins: [plugin]
    })
    expect(code).toMatchSnapshot()
  })

  test('css empty', () => {
    const basic = '(<div css=""></div>)'
    const { code } = babel.transform(basic, {
      plugins: [plugin]
    })
    expect(code).toMatchSnapshot()
  })

  test('StringLiteral css prop value', () => {
    const basic = `<div css="color: brown;"></div>`
    const { code } = babel.transform(basic, {
      plugins: [plugin]
    })
    expect(code).toMatchSnapshot()
  })

  test('noClassName', () => {
    const basic = '(<div css={`color: brown;`}></div>)'
    const { code } = babel.transform(basic, {
      plugins: [plugin]
    })
    expect(code).toMatchSnapshot()
  })

  test('emptyClassName', () => {
    const basic = '(<div className="" css={`color: brown;`}></div>)'
    const { code } = babel.transform(basic, {
      plugins: [plugin]
    })
    expect(code).toMatchSnapshot()
  })

  test('className as expression', () => {
    const basic = '(<div className={variable} css={`color: brown;`}></div>)'
    const { code } = babel.transform(basic, {
      plugins: [plugin]
    })
    expect(code).toMatchSnapshot()
  })

  test('className as expression string', () => {
    const basic =
      '(<div className={`test__class`} css={`color: brown;`} this={`hello`}></div>)'
    const { code } = babel.transform(basic, {
      plugins: [plugin]
    })
    expect(code).toMatchSnapshot()
  })

  test('no import css prop', () => {
    const basic =
      '(<div className={`test__class`} css={`color: brown;`}></div>)'
    const { code } = babel.transform(basic, {
      plugins: [[plugin, { autoImportCssProp: false }]]
    })
    expect(code).toMatchSnapshot()
  })
})
