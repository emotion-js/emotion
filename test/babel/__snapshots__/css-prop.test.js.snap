// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel css prop StringLiteral css prop value 1`] = `
"<div className={/*#__PURE__*/css([], [], function createEmotionStyledRules() {
  return [{
    \\"color\\": \\"brown\\"
  }];
})}></div>;"
`;

exports[`babel css prop basic inline 1`] = `
"<div className={\\"a\\" + \\" \\" + /*#__PURE__*/css([], [], function createEmotionStyledRules() {
  return [{
    \\"color\\": \\"brown\\"
  }];
})}></div>;"
`;

exports[`babel css prop basic object 1`] = `
"<div className={\\"a\\" + \\" \\" + css({
  \\"color\\": \\"brown\\"
})}></div>;"
`;

exports[`babel css prop basic with extractStatic 1`] = `
"import \\"./css-prop.test.emotion.css\\";
<div className={\\"a\\" + \\" \\" + \\"css-jf1v9l\\"}></div>;"
`;

exports[`babel css prop basic with extractStatic 2`] = `
".css-jf1v9l {
    color: brown
}"
`;

exports[`babel css prop className as expression 1`] = `
"<div className={variable + \\" \\" + /*#__PURE__*/css([], [], function createEmotionStyledRules() {
  return [{
    \\"color\\": \\"brown\\"
  }];
})}></div>;"
`;

exports[`babel css prop className as expression string 1`] = `
"<div className={\`test__class\` + \\" \\" + /*#__PURE__*/css([], [], function createEmotionStyledRules() {
  return [{
    \\"color\\": \\"brown\\"
  }];
})} this={\`hello\`}></div>;"
`;

exports[`babel css prop css empty 1`] = `
"<div className={/*#__PURE__*/css([], [], function createEmotionStyledRules() {
  return [{}];
})}></div>;"
`;

exports[`babel css prop dynamic inline 1`] = `
"<div className={\\"a\\" + \\" \\" + /*#__PURE__*/css([], [color], function createEmotionStyledRules(x0) {
  return [{
    \\"color\\": \`\${x0}\`
  }];
})}></div>;"
`;

exports[`babel css prop emptyClassName 1`] = `
"<div className={\\"\\" + \\" \\" + /*#__PURE__*/css([], [], function createEmotionStyledRules() {
  return [{
    \\"color\\": \\"brown\\"
  }];
})}></div>;"
`;

exports[`babel css prop no css attr 1`] = `"<div></div>;"`;

exports[`babel css prop noClassName 1`] = `
"<div className={/*#__PURE__*/css([], [], function createEmotionStyledRules() {
  return [{
    \\"color\\": \\"brown\\"
  }];
})}></div>;"
`;

exports[`babel css prop with spread arg in jsx opening tag 1`] = `
"<div className={\\"a\\" + \\" \\" + /*#__PURE__*/css([], [], function createEmotionStyledRules() {
  return [{
    \\"color\\": \\"brown\\"
  }];
})} {...rest}></div>;"
`;
